open-metadata-implementation/access-services/asset-catalog/asset-catalog-client/src/main/java/org/odpi/openmetadata/accessservices/assetcatalog/AssetCatalog.java

```diff
@@ -0,0 +1,309 @@
+        detectExceptions(methodName, response);
+
+        return response;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AssetDescriptionResponse getAssetsFromNeighborhood(String userId,
+                                                              String assetGUID,
+                                                              SearchParameters searchParameters)
+            throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, PropertyServerException {
+        String methodName = "getAssetsFromNeighborhood";
+
+        validateSearchParams(userId, assetGUID, searchParameters, methodName);
+
+        AssetDescriptionResponse response = callPostRESTCall(methodName, AssetDescriptionResponse.class,
+                serverPlatformURLRoot + BASE_PATH + ASSETS_FROM_NEIGHBORHOOD, serverName, userId, assetGUID, searchParameters);
+
+        detectExceptions(methodName, response);
+
+        return response;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AssetResponse searchByType(String userId,
+                                      String searchCriteria,
+                                      SearchParameters searchParameters)
+            throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, PropertyServerException {
+        String methodName = "searchByType";
+
+        invalidParameterHandler.validateUserId(methodName, userId);
+        invalidParameterHandler.validateSearchString(searchCriteria, "searchCriteria", methodName);
+        invalidParameterHandler.validateObject(searchParameters, SEARCH_PARAMETERS, methodName);
+
+        AssetResponse assetResponse = callPostRESTCall(methodName, AssetResponse.class,
+                serverPlatformURLRoot + BASE_PATH + SEARCH, serverName, userId, searchCriteria, searchParameters);
+
+        detectExceptions(methodName, assetResponse);
+
+        return assetResponse;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AssetResponse getAssetContext(String userId,
+                                         String assetGUID,
+                                         String assetType)
+            throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, PropertyServerException {
+        String methodName = "getAssetContext";
+
+        invalidParameterHandler.validateUserId(methodName, userId);
+        invalidParameterHandler.validateSearchString(assetGUID, GUID_PARAMETER, methodName);
+
+        AssetResponse assetResponse = callGetRESTCall(methodName, AssetResponse.class,
+                serverPlatformURLRoot + BASE_PATH + ASSET_CONTEXT, serverName, userId, assetGUID, assetType);
+
+        detectExceptions(methodName, assetResponse);
+
+        return assetResponse;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RelationshipResponse getRelationshipBetweenEntities(String userId,
```

Comment:
This method should probably return directly the relationships(similar for the other methods). This can be done in a separate PR.